See query details in `README.md`.

In summary, these numbers are PyPI download totals for Numpy wheels from 30 days prior to 7 August 2019.

In [1]:
import pandas as pd

In [2]:
df = pd.read_csv('numpy_by_pyver_distro_version.csv')
df.head(20)

downloads  python_version       distro_name distro_version
0     2576269             3.6            Ubuntu          16.04
1     2276725             3.5            Ubuntu          16.04
2     2173751             2.7  Debian GNU/Linux              9
3     1709758             2.7  Amazon Linux AMI        2018.03
4     1687085             3.6               NaN            NaN
5     1419975             2.7            Ubuntu          16.04
6     1199902             3.6  Amazon Linux AMI        2018.03
7     1148914             3.6  Debian GNU/Linux              9
8      956686             3.6            Ubuntu          18.04
9      731130             3.7               NaN            NaN
10     592500             3.7  Debian GNU/Linux              9
11     475974             2.7  Amazon Linux AMI        2017.09
12     387708             3.6      Amazon Linux              2
13     373275             3.5               NaN            NaN
14     304251             2.7            Ubuntu          14.04
15     297833             3.7  Debian GNU/Linux             10
16     291048             2.7            Ubuntu          18.04
17     250927             3.7            Ubuntu          16.04
18     243898             3.5  Debian GNU/Linux              9
19     238260             3.7            Ubuntu          18.04

In [3]:
is_python3 = df['python_version'] >= 3
df3 = df.loc[is_python3]

In [4]:
by_version = (df3.groupby(['distro_name', 'distro_version', 'python_version'])
.sum()
.sort_values('downloads', ascending=False)
             )

In [5]:
total_downloads = by_version['downloads'].sum()
total_downloads

12558832

In [6]:
by_version['download_pct'] = by_version['downloads'] / total_downloads * 100
by_version.head(20)

downloads  download_pct
distro_name      distro_version python_version                         
Ubuntu           16.04          3.6               2576269     20.513603
                                3.5               2276725     18.128477
Amazon Linux AMI 2018.03        3.6               1199902      9.554248
Debian GNU/Linux 9              3.6               1148914      9.148255
Ubuntu           18.04          3.6                956686      7.617635
Debian GNU/Linux 9              3.7                592500      4.717795
Amazon Linux     2              3.6                387708      3.087134
Debian GNU/Linux 10             3.7                297833      2.371502
Ubuntu           16.04          3.7                250927      1.998012
Debian GNU/Linux 9              3.5                243898      1.942044
Ubuntu           18.04          3.7                238260      1.897151
Debian GNU/Linux 10             3.6                232167      1.848635
Ubuntu           14.04          3.5                212657      1.693286
CentOS Linux     7              3.6                206262      1.642366
Ubuntu           14.04          3.6                169880      1.352674
Debian GNU/Linux 8              3.6                152170      1.211657
macOS            10.14.5        3.7                126053      1.003700
Amazon Linux     2              3.7                107597      0.856744
Amazon Linux AMI 2017.09        3.6                 81409      0.648221
                 2018.03        3.4                 71565      0.569838

Ubuntu 14.04 and 16.04 may well be polluted with CI runs.

I'm assuming the Debian 9 and Ubuntu 18.04 stats are persisting installs.

* Ubuntu 16.04: Python 3.5, release 4/2016, end of standard support 4/21
* Ubuntu 18.04: Python 3.6, release 4/2018, end of standard support 4/23
* Debian 9: Python 3.5, release 6/2017, EOL ~2020
* Debian 10: Python 3.7, release 7/2019, EOL ~2022

See: [Ubuntu dates](https://wiki.ubuntu.com/Releases), [Debian dates](https://wiki.debian.org/DebianReleases)

Notice Python install often differs from distribution.  For Debian 9 there are about about 8 times as many installs for (either of Pythons 3.6 or 3.7) than for the distribution standard Python 3.5.  

Some example criteria:

* Last LTS release as of 2 years ago = Python 3.6 (Ubuntu) 3.7 (Debian)
* Last LTS release as of 3 years ago = Python 3.6 (Ubuntu) 3.5 (Debian), updating to 3.7 (Debian) in 6/2020.

* Debian EOL - Python 3.5 until 2020.